compliance/virtualmachines/relay/relay.go

```diff
@@ -26,24 +26,32 @@ import (
 
 var log = logging.LoggerForModule()
 
-type VsockServer struct {
+type vsockServer interface {
+	start() error
+	stop()
+	acquireSemaphore(ctx context.Context) error
+	releaseSemaphore()
+	accept() (net.Conn, error)
+}
+
+type vsockServerImpl struct {
 	listener             *vsock.Listener
 	port                 uint32
 	semaphore            *semaphore.Weighted
 	maxSemaphoreWaitTime time.Duration
 }
 
```

Comment:
Add a check that `vsockServerImpl` does indeed implement the `vsockServer` interface:
```suggestion

var _ vsockServer = (*vsockServerImpl)(nil)

```